odl/operator/oputils.py

```diff
@@ -152,33 +159,49 @@ def power_method_opnorm(op, niter, xstart=None):
                 ValueError('a starting element must be defined in case the '
                            'operator domain has no `one()`'), exc)
     else:
-        x = xstart.copy()
+        x = op.domain.element(xstart)
 
-    tmp = op.range.element()
+    x_norm = x.norm()
+    if x_norm == 0:
+        raise ValueError('input ``x=0``')
+    x /= x_norm
+
+    niter = int(niter)
+    if niter <= 0:
+        raise ValueError('niter not non-negative')
```

Comment:
Better style: `'`foo` must be bar, got {}'.format(baz)`. More informative, and nice to have when the input is the result of another computation.